packages/components/src/query-controls/types.ts

```diff
@@ -0,0 +1,119 @@
+/**
+ * External dependencies
+ */
+import type { Key } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import type { FormTokenFieldProps } from '../form-token-field/types';
+import type { TreeSelectProps } from '../tree-select/types';
+
+export type Entity = {
+	id: number;
+	name: string;
+	parent?: number | null;
```

Comment:
`parent` [can be](https://github.com/WordPress/gutenberg/blob/921135f9ec7f1e54030672eb7d11ca28034bf15b/packages/components/src/query-controls/terms.js#L17) `null`